pkg/supervisor/supervisor_nix.go

```diff
@@ -0,0 +1,98 @@
+//go:build !windows
+// +build !windows
+
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package supervisor
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const (
+	exitCheckInterval = 200 * time.Millisecond
+)
+
+// killPid signals SIGTERM to a PID and if it's still running after
+// s.TimeoutStop sends SIGKILL.
+func (s *Supervisor) killPid(pid int) error {
+	if s.KillFunction == nil {
+		s.KillFunction = syscall.Kill
+	}
+	// Kill the process pid
+	if err := s.KillFunction(pid, syscall.SIGTERM); err != nil {
+		return fmt.Errorf("failed to send SIGTERM to pid %d: %s", pid, err)
```

Comment:
the process may already be dead, in which case kill will return [ESRCH](https://man7.org/linux/man-pages/man2/kill.2.html#ERRORS) error.